### DISCUSSION TITLE
Possibility to not use EDFA

### DISCUSSION THREAD

**Discussion by gv100204** [Telecominfraproject/oopt-gnpy#431] [Q&A] [answered]
Hi everybody,

I'd like to know if it's possible to create a link trx_A -- ROADM_A -- Fiber A-B -- ROADM_B -- trx_B?
I'd like to avoid using EDFA amplifiers, in case, for example, spans are very short.

Thanks!!

**Comment by jktjkt** [comment, marked as answer]
Yes, this is possible. GNPy will try hard to auto-add amplifiers around the ROADMs as a minimum, but this can be avoided if the topology contains a `Fused` element between the `Fiber` and the `ROADM` nodes. Alternatively, if you're really working with a linear topology and are using these physical ROADMs as muxes/demuxes and are sure that there's no amplification whatsoever in the ROADMs, perhaps you can leave them out of the simulation.

**Reply by gv100204** [reply]
hi @jktjkt 

thanks for your reply.
I can try to use Fused element and evaluate if it's good.
What do you mean when you say "you can leave them out of the simulation"? Can I remove them from a JSON file and create a connection between Fiber and Roadm or in any case I have to add a Fused element?

Thanks

**Reply by jktjkt** [reply]
I recommend to give it a try; GNPy will print out the "resulting network" after it auto-inserted amplifiers. If they are on places where they are not supposed to be at, inserting a `Fused` in their place should do the trick. Also check the resulting simulation to see if the power levels are "reasonable"; this will all be printed by GNPy.
